spark-common/src/main/js/src/spark-bootstrap.js

```diff
@@ -20,6 +20,71 @@ function AppLoader() {
         }
     };
 
+    var defaultInlineDialogOptions = {
+        width: '540px',
+        triggerText: 'Inline trigger',
+        alignment: 'bottom left'
+    };
+
+    /**
+     *
+     * @param appName Name of the application to bootstrap
+     * @param appPath application path which will be used to load the necessary angular resources
+     * @param createOptions see defaultInlineOptions (for more details - https://docs.atlassian.com/aui/7.6.3/docs/inline-dialog.html)
+     * @param startedCallback Callback which will be called after the application was successfully started
+     */
+
+    this.loadAppInInlineDialog = function(element, appName, appPath, createOptions, startedCallback) {
+
+        // append trailing slash if not there (before the query string if present)
+        var appPathParts = appPath.split('?');
```

Comment:
Move the calculation of the fullAppPath into a separate function that does the path splitting and slash handling. Also check if this function can be used in other places in this file as most of the load* functions perform related tasks.